File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/InitializeBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.*
import com.usercentrics.sdk.flutter.serializer.deserializeOptions

internal class InitializeBridge(
    private val activityProvider: FlutterActivityProvider,
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "initialize"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)

        val options = call.arguments.deserializeOptions()
        usercentrics.initialize(activityProvider.provide()?.applicationContext, options)

        result.success(null)
    }

}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/InitializeBridge.swift
import Usercentrics
import Flutter

struct InitializeBridge : MethodBridge {

    let name: String = "initialize"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)

        guard
            let options = UsercentricsOptions.initialize(from: call.arguments)
        else {
            assert(false)
            return
        }

        usercentrics.configure(options: options)
        result(nil)
    }
}
